libkbfs/kbfs_ops_test.go

```diff
@@ -995,23 +1017,14 @@ func expectSyncBlockHelper(
 	}
 	if skipSync == 0 {
 		// sign the MD and put it
-		if isUnmerged {
-			config.mockMdops.EXPECT().Put(gomock.Any(), gomock.Any()).Return(MDServerErrorConflictRevision{})
-			if rmd.BID == NullBranchID {
-				config.mockCrypto.EXPECT().MakeRandomBranchID().Return(BranchID{}, nil)
+		oldMDOps := config.MDOps()
+		if oldShim, ok := oldMDOps.(shimMDOps); ok {
+			if oldShim.isUnmerged != isUnmerged {
+				panic("old shim with different isUnmerged")
```

Comment:
Actually I guess `panic` gives you the log dump too.  But I still think a fatal would be better.